src/full/Agda/Syntax/Concrete/Definitions.hs

```diff
@@ -1211,14 +1211,12 @@ niceDeclarations fixs ds = do
             RecordDirective d -> pure [Left (fromRecordDirective d)]
             InstanceB r ds    -> do
               (dirs, ds) <- extractRecordDirectives ds
-              let ds'    = Right (InstanceB r ds)
+              let ds'    = case ds of { [] -> []; _ -> [Right (InstanceB r ds)] }
```

Comment:
`ds'` is only as much as a singleton here because this is about the one constructor of a record being declared as an instance?

Reply:
`ds` is the rest of the declarations in the instance block. You could very
much write something like this:

```agda
instance
  constructor MkR
  a : A
  a = ...

  b : B
  b = ...
```

Reply:
Aah, so it really is a list acting as a maybe. But I guess it needs to be a list later, and you wanted to avoid an unnecessary conversion?

Reply:
So what's the empty case about? You're just passing it on to get caught by an “empty instance block” warning somewhere else?

Reply:
Note that the `ds` has been shadowed: it is the (second part of the)
result of calling `extractRecordDirectives` on the original `ds`.
generating an empty instance block.

Reply:
Okay, I think I get what's going on here.